### PR TITLE
[Merged by Bors] - feat(RingTheory/Invariant): Existence of Frobenius elements

### PR DESCRIPTION
**Pull request by tb65536**
This PR proves the existence of Frobenius elements in algebraic number theory.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Invariant.lean [hunk 1]
```diff
@@ -3,23 +3,36 @@ Copyright (c) 2024 Thomas Browning. All rights reserved.
...
+import Mathlib.FieldTheory.Fixed
+import Mathlib.RingTheory.Ideal.Over
 
 /-!
-# Invariant Extensions of Rings
+# Invariant Extensions of Rings and Frobenius Elements
+
+In algebraic number theory, if `L/K` is a finite Galois extension of number fields, with rings of
+integers `𝓞L/𝓞K`, and if `q` is prime ideal of `𝓞L` lying over a prime ideal `p` of `𝓞K`, then
+there exists unique a **Frobenius element** `Frob p` in `Gal(L/K)` with the property that
```
Comment (kbuzzard) [inline Mathlib/RingTheory/Invariant.lean:14]:
```suggestion
there exists a **Frobenius element** `Frob p` in `Gal(L/K)` with the property that
```
(It's not unique in the ramified case)

File: Mathlib/RingTheory/Invariant.lean [hunk 2]
```diff
@@ -87,6 +100,212 @@ theorem isIntegral [Finite G] : Algebra.IsIntegral A B := by
     (charpoly_mem_lifts A B G b) (monic_charpoly G b)
   exact ⟨p, hp2, by rw [← eval_map, hp1, eval_charpoly]⟩
 
+/-- `G` acts transitively on primes ideals of `B` above a given prime ideal of `A`. -/
```
Comment (kbuzzard) [inline Mathlib/RingTheory/Invariant.lean:103]:
```suggestion
/-- `G` acts transitively on the prime ideals of `B` above a given prime ideal of `A`. -/
```

File: Mathlib/RingTheory/Invariant.lean [hunk 3]
```diff
@@ -87,6 +100,212 @@ theorem isIntegral [Finite G] : Algebra.IsIntegral A B := by
...
+    zero_eq_mul, or_iff_left (pow_ne_zero j ha), pow_eq_zero_iff hi, sub_eq_zero] at hf
+  exact hf.symm
+
+/-- This theorem will be made redundant by `IsFractionRing.stabilizerHom_surjective`. -/
+private theorem fixed_of_fixed1 [NoZeroSMulDivisors (B ⧸ Q) L] (f : L ≃ₐ[K] L) (b : B ⧸ Q)
+    (hx : ∀ g : MulAction.stabilizer G Q, Ideal.Quotient.stabilizerHom Q P G g b = b) :
+    f (algebraMap (B ⧸ Q) L b) = (algebraMap (B ⧸ Q) L b) := by
+  classical
+  cases nonempty_fintype G
+  revert hx
```
Comment (kbuzzard) [inline Mathlib/RingTheory/Invariant.lean:231]:
```suggestion
```
?

File: Mathlib/RingTheory/Invariant.lean [hunk 4]
```diff
@@ -87,6 +100,212 @@ theorem isIntegral [Finite G] : Algebra.IsIntegral A B := by
...
+  obtain ⟨b₀, rfl⟩ := Ideal.Quotient.mk_surjective b
+  intro hx
```
Comment (kbuzzard) [inline Mathlib/RingTheory/Invariant.lean:233]:
```suggestion
```
??

**Review by kbuzzard — APPROVED** [review]
Congratulations on getting this over the line!

bors d+

File: Mathlib/RingTheory/Invariant.lean (hunk 4 above)
Comment (tb65536) [reply Mathlib/RingTheory/Invariant.lean:233]: Guess I was a bit indecisive :)

**Pull request closed without merging**